opencensus/exporters/stats/stackdriver/internal/stackdriver_exporter.cc

```diff
@@ -103,13 +103,34 @@ void Handler::ExportViewData(
   absl::MutexLock l(&mu_);
   std::vector<google::monitoring::v3::TimeSeries> time_series;
   for (const auto& datum : data) {
-    if (!MaybeRegisterView(datum.first)) {
-      continue;
+    const opencensus::stats::ViewDescriptor& descriptor = datum.first;
+    const std::string metric_type =
+        MakeType(opts_.metric_name_prefix, descriptor.name());
+    const google::api::MonitoredResource* monitored_resource_for_view =
+        MonitoredResourceForView(descriptor, opts_.monitored_resource,
+                                 opts_.per_metric_monitored_resource);
+
+    const bool is_known_custom_metric = IsKnownCustomMetric(metric_type);
+
+    // If there is a non-default MonitoredResource for this view, it must
+    // uniquely identify the timeseries.
+    //
+    // Otherwise, we add the opencensus_task label so that different processes
+    // produce different timeseries instead of colliding.
+    const bool add_task_label =
+        is_known_custom_metric && (monitored_resource_for_view == nullptr);
```

Comment:
Is && here actually correct - I cannot quite follow the explanation.

Reply:
Updated the comment.